src/resolver.cpp

```diff
@@ -132,4 +132,10 @@ namespace libtorrent
 	{
 		m_resolver.cancel();
 	}
+
+	void resolver::set_cache_timeout(seconds timeout)
+	{
+		if (timeout >= seconds(0))
```

Comment:
I think it might be a bit confusing to ignore negative values, rather than interpreting them as 0 (or something).

Imagine I set the value to 100, then to -1. I can read the settings to say -1, but the effective timeout would still be 100.